crowbar_framework/app/models/service_object.rb

```diff
@@ -1336,6 +1340,12 @@ def apply_role(role, inst, in_queue, bootstrap = false)
         node_attr_cache[node_name]["windows"]
       end
 
+      if skip_unchanged_nodes_enabled
+        nodes_to_run = nodes_to_run.reject do |node_name|
+          skip_batch_for_node?(batch, node_name, old_role, role)
+        end
+      end
```

Comment:
I really believe you want to remove the nodes in the same way you remove nodes from the elements in the other experimental feature. This will be much more beneficial, as it allows not creating the lock files.

Reply:
mmmh, that is true, that way we dont get delayed by trying to block nodes that we are not gonna run anyway